### PR TITLE
fixes #8282: sslv3 is disabled

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

**Comment by dmitri-d** [comment]
you can test the patch by enabling ssl and executing
`nmap --script ssl-enum-ciphers -p 8443 localhost`

File: lib/smart_proxy.rb [hunk 1]
```diff
@@ -27,6 +27,94 @@
 require 'sinatra/ssl_client_verification'
 require 'sinatra/trusted_hosts'
 
+# Workaround for ruby CVE-2014-3566: creates safe defaults for SSLContext
+# also see https://www.ruby-lang.org/en/news/2014/10/27/changing-default-settings-of-ext-openssl/
```
Comment (ohadlevy) [inline lib/smart_proxy.rb]: Can we move it to another file?

**Comment by domcleal** [comment]
This should probably be configurable, it isn't the first, and won't be the last time that people will need to change ciphers or protocols in response to research.

**Comment by dmitri-d** [comment]
@domcleal: the first bit starting at line https://github.com/witlessbird/smart-proxy/blob/poodle-8282/lib/smart_proxy.rb#L36 was taken verbatim from http://svn.ruby-lang.org/cgi-bin/viewvc.cgi?revision=48121&view=revision, affects openssl default settings, and eventually (when the upstream fix makes it into most of rubys?) should be removed. Do we really care about sslv3 (or v2 for that matter) that much? There isn't a way to enable/disable a particular cipher in Webrick -- I'm not keen on introducing a custom SSL option for that, so we'll have to enable/disable this patch.

**Comment by dmitri-d** [comment]
Another approach might be not to touch ciphers at all, but disable SSLv3-related options in SSLContext, will test that now.

**Comment by dmitri-d** [comment]
Nope, ciphers need to be restricted.

**Comment by ohadlevy** [comment]
@domcleal :+1: from my side

**Comment by ohadlevy** [comment]
merged as 6ddd00d thanks @witlessbird

**Pull request closed without merging**